packages/react-native/ReactAndroid/src/test/java/com/facebook/react/uimanager/UIManagerModuleConstantsHelperTest.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.react.uimanager
+
+import com.facebook.react.common.MapBuilder
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Test
+
+class UIManagerModuleConstantsHelperTest {
+    @Test
+    fun `normalizeEventTypes withNull doesNothing`() {
+        UIManagerModuleConstantsHelper.normalizeEventTypes(null)
+    }
+
+    @Test
+    fun `normalizeEventTypes withEmptyMap doesNothing`() {
+        val emptyMap: Map<String, Any?> = MapBuilder.builder<String, Any?>().build()
+        UIManagerModuleConstantsHelper.normalizeEventTypes(emptyMap)
+        assertTrue(emptyMap.isEmpty())
+    }
+
+    @Test
+    fun `normalizeEventTypes withOnEvent doesNormalize`() {
+        val onClickMap: Map<String, Any?> =
+            MapBuilder.builder<String, Any?>().put("onClick", "¯\\_(ツ)_/¯").build()
+        UIManagerModuleConstantsHelper.normalizeEventTypes(onClickMap)
+        assertTrue(onClickMap.containsKey("topOnClick"))
+        assertTrue(onClickMap.containsKey("onClick"))
+    }
+
+    @Test
+    fun `normalizeEventTypes withTopEvent doesNormalize`() {
+        val onClickMap: Map<String, Any?> =
+            MapBuilder.builder<String, Any?>().put("topOnClick", "¯\\_(ツ)_/¯").build()
+        UIManagerModuleConstantsHelper.normalizeEventTypes(onClickMap)
+        assertTrue(onClickMap.containsKey("topOnClick"))
+        assertFalse(onClickMap.containsKey("onClick"))
+    }
+
+    @Test
+    fun `normalizeEventTypes withNestedObjects doesNotLoseThem`() {
+        val nestedObjects: Map<String, Any?> =
+            MapBuilder.builder<String, Any?>()
+                .put(
+                    "onColorChanged",
+                    MapBuilder.of<String, Any?>(
+                        "phasedRegistrationNames",
+                        MapBuilder.of<String, Any?>(
+                            "bubbled",
+                            "onColorChanged",
+                            "captured",
+                            "onColorChangedCapture"
+                        )
+                    )
+                )
+                .build()
+        UIManagerModuleConstantsHelper.normalizeEventTypes(nestedObjects)
+        assertTrue(nestedObjects.containsKey("topOnColorChanged"))
+        @Suppress("UNCHECKED_CAST")
```

Comment:
Makes sense. I believe using [`assertNotNull`](https://kotlinlang.org/api/latest/kotlin.test/kotlin.test/assert-not-null.html) from kotlin.test instead of junit would also fix this. What are your thoughts on potentially adding kotlin.test as a dependency.

Reply:
Nope I don't think it will fix this. The problem here are all the `as? Map<String, Any?>` cast that can't easily be avoided. We should desing a better `MapBuilder` that is more Kotlin friendly.

Reply:
Ah gotcha. I thought maybe the [use of contract](https://github.com/JetBrains/kotlin/blob/924c28507067cbfbf78a6509ea89eabe496e34ca/libraries/kotlin.test/common/src/main/kotlin/kotlin/test/Assertions.kt#L144) in there would fix this particular warning but definitely fixing the root of the problem and avoiding the need to cast altogether would be better.